tensorboard/webapp/metrics/store/metrics_selectors.ts

```diff
@@ -495,29 +515,47 @@ export const getMetricsCardMinMax = createSelector(
  */
 export const getMetricsCardTimeSelection = createSelector(
   getCardStateMap,
+  getMetricsStepSelectorEnabled,
   getMetricsLinkedTimeEnabled,
   getMetricsLinkedTimeSelection,
   (
     cardStateMap: CardStateMap,
+    globalStepSelectionEnabled: boolean,
     linkedTimeEnabled: boolean,
     linkedTimeSelection: TimeSelection | null,
     cardId: CardId
   ): TimeSelection | undefined => {
+    // Handling Linked Time
     if (linkedTimeEnabled && linkedTimeSelection) {
       return linkedTimeSelection;
     }
 
-    if (cardStateMap[cardId]?.timeSelection) {
-      return cardStateMap[cardId]?.timeSelection;
+    const cardState = cardStateMap[cardId];
+    if (!cardState) {
+      return;
     }
-    const minMaxStep = getMinMaxStepFromCardState(cardStateMap[cardId]);
+    const minMaxStep = getMinMaxStepFromCardState(cardState);
     if (!minMaxStep) {
       return;
     }
 
+    // If the user has disabled step selection, nothing should be returned.
+    if (
+      !cardSelectionStateToBoolean(
+        cardState.stepSelection,
+        globalStepSelectionEnabled
+      )
+    ) {
+      return;
+    }
+
+    const startStep = cardState.timeSelection?.start.step ?? minMaxStep.minStep;
+    const endStep = cardState.timeSelection?.end?.step ?? minMaxStep.maxStep;
```

Comment:
Ensuring that an end step is always set allows the responsibility of rending the end step to be delegated to the view. This also makes refreshing the data work more cleanly after only a single step has been set.